src/events/kyuubi.py

```diff
@@ -61,6 +71,18 @@ def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
             zookeeper_info=self.context.zookeeper,
         )
 
+        # Check the newly created service is connectable
+        if not self.service_util.is_service_connectable():
```

Comment:
On `config-changed` event, we reconcile the required external service and immediately check whether the service is available for connection. If not, we simply defer this config-changed event in the hopes that when the deferred event comes back later, the service endpoint will be available.

It is to be noted that when the deferred event comes, `recoincile_services` will not delete and recreate the external service, it just checks and finds that the desired service is already created and return.